test/Test/TypeRep/TypeRepMapProperty.hs

```diff
@@ -150,3 +162,29 @@ genTF = do
     case someNatVal randNat of
         Just (SomeNat proxyNat) -> pure $ WrapTypeable $ IntProxy proxyNat randInt
         Nothing                 -> error "Invalid test generator"
+
+----------------------------------------------------------------------------
+-- Helpers
+----------------------------------------------------------------------------
+#if __GLASGOW_HASKELL__ < 806
+{- | This newtype is used to compare 'TypeRepMap's using only 'Fingerprint's.
+It's not a good idea to write such 'Eq' instance for 'TypeRepMap' itself because
+it doesn't compare values so it's not true equality. But this should be enough
+for tests.
+-}
+newtype FpMap f = FpMap (TypeRepMap f)
+    deriving newtype (Show, Semigroup, Monoid)
+
+instance Eq (FpMap f) where
+    FpMap (TypeRepMap as1 bs1 _ _) == FpMap (TypeRepMap as2 bs2 _ _) =
+        as1 == as2 && bs1 == bs2
+
+-- Define an wrapper around equality checking for TRMs that works on both
+-- old and new GHCs.
+(!===!) :: (MonadTest m, HasCallStack) => TypeRepMap f -> TypeRepMap f -> m ()
```

Comment:
Rather than litter the whole file with CPP I figured it might be easier to overload a single operator that just "does the right thing" since many future property tests are likely to need equality testing :)

Let me know what you think about that, happy to switch to an infix `trmEqual` or something if that's better.

Reply:
@ChrisPenner Good idea! And I think, there's even a simpler solution for this problem. You can define an orphan instance for `TypeRepMap` directly only in this module that compares values only by `Fingerpint`s.

With this approach, we don't need to introduce `newtype` and a custom operator at all, we can use just `===`. Since it's only for testing, I think this is okay. And since we are not going to support old GHC versions forever, this orphan instance will be removed eventually in some future.